reference/events.rst

```diff
@@ -0,0 +1,189 @@
+Symfony Framework Events
+========================
+
+Kernel Events
+-------------
+
+Each event thrown by the Kernel is a subclass of
+:class:`Symfony\\Component\\HttpKernel\\Event\\KernelEvent`. This means that
+each event has access to the following information:
+
+:method:`Symfony\\Component\\HttpKernel\\Event\\KernelEvent::getRequestType`
+    Returns the *type* of the request (``HttpKernelInterface::MASTER_REQUEST`` or
+    ``HttpKernelInterface::SUB_REQUEST``).
+
+:method:`Symfony\\Component\\HttpKernel\\Event\\KernelEvent::getKernel`
+    Returns the Kernel handling the request.
+
+:method:`Symfony\\Component\\HttpKernel\\Event\\KernelEvent::getRequest`
+    Returns the current ``Request`` being handled.
+
+.. _kernel-core-request:
+
+``kernel.request`` Event
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+*Event Class*: :class:`Symfony\\Component\\HttpKernel\\Event\\GetResponseEvent`
+
+The goal of this event is to either return a ``Response`` object immediately
+or setup variables so that a Controller can be called after the event. Any
+listener can return a ``Response`` object via the ``setResponse()`` method on
+the event. In this case, all other listeners won't be called.
+
+This event is used by the FrameworkBundle to populate the ``_controller``
+``Request`` attribute, via the
+:class:`Symfony\\Bundle\\FrameworkBundle\\EventListener\\RouterListener`.
+RequestListener uses a :class:`Symfony\\Component\\Routing\\RouterInterface`
+object to match the ``Request`` and determine the Controller name (stored in the
+``_controller`` ``Request`` attribute).
+
+.. seealso::
+
+    Read more on the :ref:`kernel.request event <component-http-kernel-kernel-request>`.
+
+``kernel.controller`` Event
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*Event Class*: :class:`Symfony\\Component\\HttpKernel\\Event\\FilterControllerEvent`
+
+This event is not used by the FrameworkBundle, but can be an entry point used
```

Comment:
instead of showing an example here, I think we can link to http://symfony.com/doc/current/cookbook/event_dispatcher/before_after_filters.html